policies/recipes/steps.go

```diff
@@ -396,12 +397,32 @@ func stepInstallMsi(step *agentendpointpb.SoftwareRecipe_Step_InstallMsi, artifa
 	return executeCommand("C:\\Windows\\System32\\msiexec.exe", args, stepDir, runEnvs, exitCodes)
 }
 
-func stepInstallDpkg(step *agentendpointpb.SoftwareRecipe_Step_InstallDpkg, artifacts map[string]string, runEnvs []string, stepDir string) error {
-	return fmt.Errorf("InstallDpkg not yet supported")
+func stepInstallDpkg(step *agentendpointpb.SoftwareRecipe_Step_InstallDpkg, artifacts map[string]string) error {
+	if !packages.DpkgExists {
+		return fmt.Errorf("dpkg does not exist on system")
+	}
+
+	artifact := step.GetArtifactId()
+	path, ok := artifacts[artifact]
+	if !ok {
+		return fmt.Errorf("%q not found in artifact map", artifact)
+	}
+
+	return packages.DpkgInstall(path)
 }
 
-func stepInstallRpm(step *agentendpointpb.SoftwareRecipe_Step_InstallRpm, artifacts map[string]string, runEnvs []string, stepDir string) error {
-	return fmt.Errorf("InstallRpm not yet supported")
+func stepInstallRpm(step *agentendpointpb.SoftwareRecipe_Step_InstallRpm, artifacts map[string]string) error {
+	if !packages.RPMExists {
+		return fmt.Errorf("rpm does not exist on system")
```

Comment:
is this error wrapped by step runner? i think the full error message needs to be more like 'InstallRPM step but `rpm` does not exist on system'

Reply:
Added better error logging at the caller.